src/Compilers/CSharp/Portable/Binder/Binder_Expressions.cs

```diff
@@ -5842,6 +5843,18 @@ private BoundExpression BindCheckedExpression(CheckedExpressionSyntax node, Diag
         /// cannot.  If this is not a Color Color case, then return null.
         /// </summary>
         private BoundExpression BindLeftOfPotentialColorColorMemberAccess(ExpressionSyntax left, DiagnosticBag diagnostics)
+        {
+            if (left is IdentifierNameSyntax identifier)
+            {
+                return BindLeftIdentifierOfPotentialColorColorMemberAccess(identifier, diagnostics);
+            }
+
+            // NOTE: it is up to the caller to call CheckValue on the result.
+            return BindExpression(left, diagnostics);
+        }
+
+        [MethodImpl(MethodImplOptions.NoInlining)]
+        private BoundExpression BindLeftIdentifierOfPotentialColorColorMemberAccess(IdentifierNameSyntax left, DiagnosticBag diagnostics)
```

Comment:
Consider leaving a comment on why we have a NoInlining here, like we do for the `VisitRValue` code in `NullableWalker.`